atlasdb-tests-shared/src/test/java/com/palantir/atlasdb/sweep/queue/TargetedSweepTest.java

```diff
@@ -313,4 +316,25 @@ public void cleanup() {}
     private void sweepNextBatch(ShardAndStrategy shardStrategy) {
         sweepQueue.sweepNextBatch(shardStrategy, Sweeper.of(shardStrategy).getSweepTimestamp(sweepTimestampSupplier));
     }
+
+    /**
+     * After this method returns successfully, we know that there has existed some point in time when the immutable
+     * timestamp from the {@link #serializableTxManager} was strictly greater than the timestamp parameter that was
+     * passed. In the absence of concurrent transaction starts, this also means that the immutable timestamp will be
+     * strictly greater than the timestamp parameter going forward.
+     *
+     * Note that in the general case, this does not guarantee that the immutable timestamp is currently greater than
```

Comment:
Reads more complex - can we simplify this a bit?  like explicitly have sections like 
1. what happens
2. why that is okay
3. why we need to work around it for tests 
4. how we deal with this in tests

Reply:
I think splitting into sections is OK (e.g. based on paragraphs, an outline of what you describe is fine) though explicit sections isn't consistent with our Javadoc style in general.